### PR TITLE
HEV Zombies are separate entity class from Zombine

### PR DESCRIPTION
**Pull request by 1upD**
npc_zombine with the Xen EZ variant are presented as HEV suit zombies from the Black Mesa survey team. This PR will split that NPC into its own new class, npc_hevzombie. npc_hevzombie supports multiple variants and by default will randomly select whether or not to spawn with a held crowbar. This is based on the metrozombie for EZU.

### CODE REVIEW

File: sp/src/game/server/hl2/npc_zombine.cpp [hunk 1]
```diff
@@ -1802,6 +1845,380 @@ void CNPC_MetroZombie::ReadyGrenadeSound( void )
...
+//-----------------------------------------------------------------------------
+void CNPC_HEVZombie::Event_Killed( const CTakeDamageInfo &info )
+{
+	BaseClass::Event_Killed( info );
+
+	if (m_bDropItems) {
+		int numBatteries = MAX( m_ArmorValue / sk_battery.GetInt(), random->RandomInt( 1, 3 ));
+		for (int i = 0; i < numBatteries; i++)
+		{
+			DropItem( "item_battery", WorldSpaceCenter() + RandomVector( -4, 4 ), RandomAngle( 0, 360 ) );
```
Comment (Blixibon) [inline sp/src/game/server/hl2/npc_zombine.cpp:2216]: This should match the HEV zombie's E:Z variant

File: sp/src/game/server/hl2/npc_zombine.cpp [hunk 2]
```diff
@@ -1802,6 +1845,380 @@ void CNPC_MetroZombie::ReadyGrenadeSound( void )
...
+
+	BaseClass::Precache();
+}
+
+void CNPC_HEVZombie::ChooseDefaultGrenadeType()
+{
+	int chance = random->RandomInt( 0, 3 );
+
+	// If we set the type to "Arbeit", give them Xen grenades
+	if (m_tEzVariant == EZ_VARIANT_ARBEIT)
```
Comment (Blixibon) [inline sp/src/game/server/hl2/npc_zombine.cpp]: Shouldn't the rad variant also be given Xen grenades? 🤔

File: sp/src/game/server/hl2/npc_zombine.cpp (hunk 1 above)
Comment (1upD) [reply sp/src/game/server/hl2/npc_zombine.cpp:2216]: Doesn't DropItem() do that already? I thought it was overridden in basenpc

File: sp/src/game/server/hl2/npc_zombine.cpp (hunk 2 above)
Comment (1upD) [reply sp/src/game/server/hl2/npc_zombine.cpp]: Perhaps... That seems like a good idea to me

File: sp/src/game/server/hl2/npc_zombine.cpp (hunk 1 above)
Comment (Blixibon) [reply sp/src/game/server/hl2/npc_zombine.cpp:2216]: Whoops! You're right, sorry 😅

**Review by Blixibon — APPROVED** [review]

**Pull request merged**